scipy/signal/_peak_finding.py

```diff
@@ -233,6 +236,733 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
+
+    keep = ~del_flag[np.argsort(eval_peaks)]
+
+    return keep
+
+
+def find_peaks(x, height=None, threshold=None, distance=None,
+               prominence=None, width=None, wlen=None, rel_height=0.5):
+    """
+    Find a subset of peaks inside a signal.
+
+    This function takes a one-dimensional array and finds all local maxima by
+    simple comparision of neighbouring values. A subset of these peaks can be
+    selected by specifying requirements (filter options) for a peak's properties.
+
+    Parameters
+    ----------
+    x : sequence
+        A signal with peaks.
+    height : number or ndarray or sequence, optional
+        Required height of peaks. Either a number, ``None``, an array matching
+        `x` or a 2-element sequence of the former. The first element is
+        always interpreted as the  minimum and the second, if supplied, as the
+        maximum required height.
+    threshold : number or ndarray or sequence, optional
+        Required threshold of peaks, the vertical distance to its neighbouring
+        samples. Either a number, ``None``, an array matching `x` or a
+        2-element sequence of the former. The first element is always
+        interpreted as the  minimum and the second, if supplied, as the maximum
+        required threshold.
+    distance : number, optional
+        Required minimal horizontal distance (>= 1) in samples between
+        neighbouring peaks. The removal order is explained in the notes section.
+    prominence : number or ndarray or sequence, optional
+        Required prominence of peaks. Either a number, ``None``, an array
+        matching `x` or a 2-element sequence of the former. The first
+        element is always interpreted as the  minimum and the second, if
+        supplied, as the maximum required prominence.
+    width : number or ndarray or sequence, optional
+        Required width of peaks. Either a number, ``None``, an array
+        matching `x` or a 2-element sequence of the former. The first
+        element is always interpreted as the  minimum and the second, if
+        supplied, as the maximum required prominence.
+    wlen : number, optional
+        Used for calculation of the peaks prominences thus it is only used if
+        one of the arguments `prominence` or `width` is given. See argument
+        `wlen` in `peak_prominences` for a full description of its effects.
+    rel_height : float, optional
+        Used for calculation of the peaks width thus it is only used if `width`
+        is given. See argument  `rel_height` in `peak_widths` for a full
+        description of its effects.
+
+    Returns
+    -------
+    peaks : ndarray
+        Indices of peaks in `x` that satisfy all requirements.
+    properties : dict
+        A dictionary containing properties of the returned peaks which were
+        calculated as intermediate results during filtering:
+
+        * If `height` is given, the value of `x` at each peak is returned
```

Comment:
Could you try formatting as a RST definition list like:
```
... during filtering, with keys:

* 'peak_heights'
      If 'height' is given, the value of `x` at each peak.
* ...
```
I think it will render nicely and make it easier to identify the keys.